tests/drivers/uart/uart_async_api/boards/nucleo_f746zg.conf

```diff
@@ -0,0 +1 @@
+CONFIG_NOCACHE_MEMORY=y
```

Comment:
Would it be interesting to have an error or a warning at kconfig level if application doesn't enable this switch (in case of DMA and cache enabled) ?
Message could explain that DMA buffer should be placed in the no cache memory.